server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/DiskBBQBulkWriter.java

```diff
@@ -29,6 +29,22 @@ protected DiskBBQBulkWriter(int bulkSize, IndexOutput out) {
         this.out = out;
     }
 
+    /**
+     * Bulk write vectors to disk, optionally also block encoding tail vectors (batches smaller than {@code bulkSize}).
+     * @param qvv quantized vector values
+     * @param docsWriter docs writer
+     * @param encodeAll whether to block encode tail vectors
+     * @throws IOException if writing fails
+     */
+    public abstract void writeVectors(QuantizedVectorValues qvv, CheckedIntConsumer<IOException> docsWriter, boolean encodeAll)
```

Comment:
I am wondering if instead of adding this boolean at the end , we should add new BulkWriters implementations. We will make this class polymorphic but I think it does not matter here. This is call once per cluster and it is not so performance sensitive so metamorphic calls here are ok.